Cython/Compiler/TypeSlots.py

```diff
@@ -268,6 +268,10 @@ def generate_spec(self, scope, code):
         if value == "0":
             return
         preprocessor_guard = self.preprocessor_guard_code()
+        if not preprocessor_guard:
+            if self.py3 and self.slot_name.startswith('bf_'):
```

Comment:
@scoder Although unrelated to buffers, maybe we should extend
```python
    prefix = self.slot_name[:3]
    if self.py3 and (prefix == 'bf_' or prefix == 'am_'):
        ...
```
This will account for the addition of `Py_am_send` to the limited API v3.10.

```diff
$ diff /usr/include/python3.9/typeslots.h  /usr/include/python3.10/typeslots.h 
90a91,94
> #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x030A0000
> /* New in 3.10 */
> #define Py_am_send 81
> #endif
```